File: Pythonleetcode/Leetcode1732.py
class Solution(object):
    def largestAltitude(self, gain):
        ans=[0]
        for i in gain:
            ans.append(ans[-1]+i)
        return max(ans)

print(Solution().largestAltitude( [-5,1,5,0,-7]))
print(Solution().largestAltitude( [-4,-3,-2,-1,4,3,2]))

File: CppPractice/Project1/Project1/Leetcode1732.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

#include <map>

using namespace std;

class Solution {
public:
    int largestAltitude(vector<int>& gain) {
		int ans=0;
		int temp_h=0;
		for (int i = 0; i < gain.size(); i++)
		{
			temp_h += gain[i];
			ans = max(ans, temp_h);

		}
		return ans;

    }
};